File: 0141-linked-list-cycle/0141-linked-list-cycle.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public bool HasCycle(ListNode head) {
        
        var hs = new HashSet<ListNode>();
        
        while(head != null){
            if(!hs.Add(head.next)) return true;
            head = head.next;
        }
        
        return false;
    }
}

File: 0141-linked-list-cycle/0141-linked-list-cycle.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function hasCycle(head: ListNode | null): boolean {
    if(head === null) return false;
    
    let slow: ListNode = head;
    let fast: ListNode = head.next;
    
    while(slow != null && fast != null){
        if(slow === fast) return true;
        slow = slow.next;
        fast = fast.next?.next;
    }
        
    return false;
};